documents/sim-objects/branch-predictors.md

```diff
@@ -0,0 +1,76 @@
+# Branch Predictors
+
+## Branch Predictors in gem5
+1. 2-bit local
+2. Bi-Mode
+3. Tournament
+4. Loop predictor
+5. TAGE
+6. Statistical Corrector
+7. Multi Perspective Perceptron predictor
+
+### Local Predictor
+The local predictor is a 1 level branch predictor that has a history table in which each entry is a n bit saturating counter. The history table is indexed by the PC, each counter in history table has 2^N states, if it predicts taken, branch target table in the indirector predictor is used obtain the branch address.  
```

Comment:
Some proofreading:
> each entry is a n bit saturating counter
each entry is an *N-bit* saturating counter
> each counter in history table
each counter in *the* history table
> branch target table in the indirector predictor is used obtain the branch address
the branch target table in the *indirect* predictor is used *to* ...